packages/synapse-react-client/src/components/GoalsV2/GoalsV2.tsx

```diff
@@ -0,0 +1,226 @@
+  countSql?: string
+  title: string
+  summary: string
+  link: string
+  asset: string
+}
+
+enum ExpectedColumns {
+  TABLEID = 'TableId', // Both TableId or CountSql are used to indicate what Table rows to count.
+  COUNT_SQL = 'CountSql', // Code uses CountSql over TableId if defined (if the CountSql column is in the schema and filled in).
+  TITLE = 'Title',
+  SUMMARY = 'Summary',
+  LINK = 'Link',
+  ASSET = 'Asset',
+}
+
+// PORTALS-2367
+const GOALSV2_DESKTOP_MIN_BREAKPOINT = 1200
+
+export const GoalsV2: React.FC<GoalsV2Props> = (props: GoalsV2Props) => {
+  const { entityId, dataLink } = props
+  const { accessToken } = useSynapseContext()
+  const [assets, setAssets] = useState<string[] | undefined>()
+  const [error, setError] = useState<string | SynapseClientError | undefined>()
+  const showDesktop = useShowDesktop(GOALSV2_DESKTOP_MIN_BREAKPOINT)
+  const queryBundleRequest: QueryBundleRequest = {
+    concreteType: 'org.sagebionetworks.repo.model.table.QueryBundleRequest',
+    entityId,
+    partMask:
+      SynapseConstants.BUNDLE_MASK_QUERY_SELECT_COLUMNS |
+      SynapseConstants.BUNDLE_MASK_QUERY_RESULTS,
+    query: {
+      sql: `select * from ${entityId} order by ItemOrder`,
+    },
+  }
+  const { data: queryResultBundle } =
+    useGetQueryResultBundle(queryBundleRequest)
+
+  useEffect(() => {
+    const getData = async () => {
+      try {
+        const assetColumnIndex = getFieldIndex(
+          ExpectedColumns.ASSET,
+          queryResultBundle,
+        )
+        const assets = (queryResultBundle?.queryResult!.queryResults.rows.map(
+          el => el.values[assetColumnIndex],
+        ) ?? []) as string[]
+        const assetFileHandleIds = assets.filter(
+          v => v != null && v !== undefined,
+        )
+        if (assetFileHandleIds.length === 0) {
+          // wait for data to load
+          return
+        }
+        const fileHandleAssociationList: FileHandleAssociation[] =
+          assetFileHandleIds.map(fileId => {
+            return {
+              associateObjectId: entityId,
+              associateObjectType: FileHandleAssociateType.TableEntity,
+              fileHandleId: fileId,
+            }
+          })
+        const batchFileRequest: BatchFileRequest = {
+          includeFileHandles: false,
+          includePreSignedURLs: true,
+          includePreviewPreSignedURLs: false,
+          requestedFiles: fileHandleAssociationList,
+        }
+        const files = await getFiles(batchFileRequest, accessToken)
+        setError(undefined)
+        {
+          /* TODO - handle case where presigned URL expires */
+        }
+        setAssets(
+          files.requestedFiles
+            .filter(el => el.preSignedURL !== undefined)
+            .map(el => el.preSignedURL!),
+        )
+      } catch (e) {
+        console.error('Error on get data', e)
+        setError(e)
+      }
+    }
+    getData()
+  }, [entityId, accessToken, queryResultBundle])
```

Comment:
Is the logic identical between this in GoalsV1 and V2? If so, consider extracting a custom hook used in both GoalsV1 and V2